### DISCUSSION TITLE
RegexManager and Docker digest pinning - Help to make it work for the 1st time

### DISCUSSION THREAD

**Discussion by xakraz** [renovatebot/renovate#15238] [Request Help] [answered]
Hi there!

Related to https://github.com/renovatebot/renovate/discussions/15005, I have an other question regarding Regex manager and Docker digest updates.

## Context

We currently have repos that do not pin the digest of their docker images.

While pocing Renovate on 1 repo, I used the default `dockerfile` manager, and Renovate opened PRs to pin the digest of the images.
So far, so good.

Except that, for misc reasons, we have put the docker registry in an `ARG` variable.
So I had to set up the `regexManagers` to be able to parse the specific `FROM` statement and set the registry URL for the lookup part.

```
{
  "extends": [
    "config:base",
    ":disableDependencyDashboard",
    ":docker",
    "docker:pinDigests",
    "docker:disableMajor"
  ],

  "enabledManagers": [
    // "dockerfile",
    "regex"
  ],


  "regexManagers": [
    {
      "fileMatch": [
        "(^|/|\\.)Dockerfile$",
        "(^|/)Dockerfile\\.[^/]*$"
      ],
      "matchStrings": [
        "FROM \\$\\{DOCKER_REGISTRY\\}\/(?<repo>[a-z0-9-\/]+):(?<currentValue>[a-z0-9-]+)(@(?<currentDigest>sha256:[a-f0-9]+))?"
      ],
      "depNameTemplate": "us.gcr.io/{{{repo}}}",
      "datasourceTemplate": "docker",
      "versioningTemplate": "docker",
    }
  ],

  "packageRules": [
    {
      "matchDatasources": ["docker"],
      "matchUpdateTypes": [
        "minor",
        "patch",
        "pin",
        "digest"
      ],
      "enabled": true,

      // GitHub Automation
      "addLabels": ["dependencies", "docker"],
      "automerge": true,
      "automergeType": 'pr',
      "platformAutomerge": true,
    },
    {
      "matchDatasources": ["docker"],
      "matchUpdateTypes": [
        "major"
      ],
      "enabled": false
    }
  ]
}
```

### Note 1

* While applying this **updated config to my POC repo**, it worked as expected. Tags and digest get updated.
* However, while applying **this config on a new repo**, it doesn't pin digest and only suggests tags update.

### Note 2

If I manually pin the digest in the Dockerfile of the "new repo", the next Renovate run does what is expected: Tags and digest update through PRs.

## Question

What am I missing so that Renovate can "onboard" repos to pinned digest images without us doing it manually first?

Thanks 🙏🏻

**Comment by viceice** [comment, marked as answer]
not with regex manager, sorry.

**Reply by alexrecuenco** [reply]
Given the answer in https://github.com/renovatebot/renovate/issues/10993#issuecomment-2243203454, what do you think is the right solution?

**Comment by viceice** [comment]
also the `docker:` presets probably won't work as expected, as they match on docker only managers.
